clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -93,81 +135,114 @@ func (a *apiAvailabilityMeasurement) start(config *measurement.Config, SSHToMast
 			select {
 			case <-a.stopCh:
 				return
-			case <-time.After(time.Duration(probeDuration)):
+			case <-a.holdCh:
+				select {
+				case <-a.stopCh:
+					return
+				case <-a.holdCh:
+					continue
+				}
+			case <-time.After(a.frequency):
 				a.updateClusterAvailabilityMetrics(k8sClient)
-				if SSHToMasterSupported {
-					a.updateMasterAvailabilityMetrics(k8sClient, config, provider)
+				if a.masterLevelAvailabilityEnabled() {
+					a.updateMasterAvailabilityMetrics(k8sClient, provider)
 				}
 			}
 		}
 	}()
 	return nil
 }
 
-// Execute starts the api-server healthz probe end point from start action and
-// collects availability metrics in gather.
-func (a *apiAvailabilityMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
-	SSHToMasterSupported := !config.ClusterFramework.GetClusterConfig().Provider.Features().SupportSSHToMaster
-
-	if !SSHToMasterSupported {
-		klog.Infof("ssh to master nodes not supported. Measurement would have only Cluster Level Metrics")
+func (a *apiAvailabilityMeasurement) initFields(config *measurement.Config) error {
+	a.isRunning = true
+	a.stopCh = make(chan struct{})
+	a.holdCh = make(chan struct{})
+	frequency, err := util.GetDuration(config.Params, "frequency")
+	if err != nil {
+		return err
+	}
+	a.frequency = frequency
+
+	a.clusterLevelMetrics = &apiAvailabilityMetrics{}
+	if config.ClusterLoaderConfig.EnableExecService {
```

Comment:
exec service is [disabled by default](https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/cmd/clusterloader.go#L116), just fyi